merlin/models/tf/transforms/tensor.py

```diff
@@ -101,33 +101,48 @@ def __init__(self, max_seq_length: Optional[int] = None, **kwargs):
         super().__init__(**kwargs)
         self.max_seq_length = max_seq_length
 
-    def call(self, inputs: TabularData, **kwargs) -> TabularData:
+    def call(self, inputs: Union[tuple, tf.RaggedTensor, TabularData], **kwargs) -> TabularData:
+        if isinstance(inputs, (tf.RaggedTensor, tuple)):
+            return self._convert_tensor_to_dense(inputs)
+
         outputs = {}
         for name, val in inputs.items():
-            if isinstance(val, tuple):
-                val = list_col_to_ragged(val)
-            if isinstance(val, tf.RaggedTensor):
-                if self.max_seq_length:
-                    outputs[name] = val.to_tensor(shape=[None, self.max_seq_length])
-                else:
-                    outputs[name] = tf.squeeze(val.to_tensor())
-            else:
-                outputs[name] = tf.squeeze(val)
-
+            outputs[name] = self._convert_tensor_to_dense(val)
         return outputs
 
     def compute_output_shape(self, input_shape):
-        batch_size = self.calculate_batch_size_from_input_shapes(input_shape)
-        outputs = {}
+        if not isinstance(input_shape, dict):
+            return self._get_output_tensor_shape(input_shape)
 
+        outputs = {}
         for key, val in input_shape.items():
-            if isinstance(val, tuple):
-                outputs[key] = tf.TensorShape((batch_size, self.max_seq_length))
-            else:
-                outputs[key] = tf.TensorShape((batch_size))
-
+            outputs[key] = self._get_output_tensor_shape(val)
         return outputs
 
+    def _get_output_tensor_shape(self, val_shape):
+        if isinstance(val_shape, tuple) and isinstance(val_shape[1], tf.TensorShape):
+            val_shape = val_shape[1]
+            return tf.TensorShape([val_shape[0], self.max_seq_length])
+        if val_shape.rank > 1 and val_shape[-1] > 1:
+            shapes = val_shape.as_list()
+            if self.max_seq_length:
+                shapes[1] = self.max_seq_length
+            return tf.TensorShape(shapes)
+        return tf.TensorShape((val_shape[0]))
+
+    def _convert_tensor_to_dense(self, val):
+        if isinstance(val, tuple):
+            val = list_col_to_ragged(val)
+        if isinstance(val, tf.RaggedTensor):
+            if self.max_seq_length:
+                shape = [None] * val.shape.rank
+                shape[1] = self.max_seq_length
+                val = val.to_tensor(shape=shape)
+            else:
+                val = tf.squeeze(val.to_tensor())
```

Comment:
It's not clear to me why calling squeeze is related to conversion from ragged to dense. However, this was already here so maybe something to review another time.

Reply:
Right, I suspect it is needed to remove any trailing dimension =`1` we can get from the data loader's multi-hot outputs but I couldn't reproduce this specific case. I also removed `tf.squeeze` and re-run the tests, all of them passed.